pkg/tftarget/tftarget_test.go

```diff
@@ -0,0 +1,270 @@
+		match:     match(excludeAddress("google_compute_global_forwarding_rule.test")),
+		address:   "google_compute_global_forwarding_rule.test",
+		wantMatch: false,
+	},
+	{
+		name:      "exclude does not match org for nested module",
+		match:     match(excludeAddress("google_compute_global_forwarding_rule.test")),
+		address:   "module.abc.google_compute_global_forwarding_rule.test",
+		wantMatch: true,
+	},
+	{
+		name:      "exclude name wildcard match",
+		match:     match(excludeAddress("google_compute_global_forwarding_rule.*")),
+		address:   "google_compute_global_forwarding_rule.test",
+		wantMatch: false,
+	},
+	{
+		name:      "exclude module wildcard match",
+		match:     match(excludeAddress("**.google_compute_global_forwarding_rule.*")),
+		address:   "module.abc.google_compute_global_forwarding_rule.test",
+		wantMatch: false,
+	},
+	// errors
+	{
+		name:                "nested spaces",
+		match:               match(includeAddress("**.* *.*")),
+		address:             "module.abc.google_compute_global_forwarding_rule.test",
+		wantConstraintError: true,
+	},
+	{
+		name:                "nested special characters",
+		match:               match(includeAddress("**$$")),
+		address:             "module.abc.google_compute_global_forwarding_rule.test",
+		wantConstraintError: true,
+	},
+	{
+		name:                "exclude error - nested spaces",
+		match:               match(excludeAddress("**. * *.*")),
+		address:             "module.abc.google_compute_global_forwarding_rule.test",
+		wantConstraintError: true,
+	},
+	{
+		name:                "exclude error - nested special characters",
+		match:               match(excludeAddress("**$$")),
+		address:             "module.abc.google_compute_global_forwarding_rule.test",
+		wantConstraintError: true,
+	},
+	{
+		name:         "provider contains google",
+		address:      "google_compute_global_forwarding_rule.test",
+		wantMatch:    true,
+		providerName: "google",
+	},
+	{
+		name:         "provider doesn't contain google",
+		address:      "google_compute_global_forwarding_rule.test",
+		wantMatch:    false,
+		providerName: "moogle",
+	},
+
+	{
+		name:                "provider block missing",
+		address:             "google_compute_global_forwarding_rule.test",
+		wantMatch:           true,
+		removeProviderBlock: true,
+	},
+
+	{
+		name: "Bad target type",
+		match: map[string]interface{}{
+			"addresses": "%#*(*#$(#$)",
+		},
+		wantConstraintError: true,
+	},
+	{
+		name: "Bad target item type",
+		match: map[string]interface{}{
+			"addresses": []interface{}{1},
+		},
+		wantConstraintError: true,
+	},
+	{
+		name: "Bad exclude type",
+		match: map[string]interface{}{
+			"excludedAddresses": ")$(*#$)*$*&#x",
+		},
+		wantConstraintError: true,
+	},
+	{
+		name: "Bad exclude item type",
+		match: map[string]interface{}{
+			"excludedAddresses": []interface{}{1},
+		},
+		wantConstraintError: true,
+	},
+}
+
+func TestTargetHandler(t *testing.T) {
```

Comment:
Since most of this logic is ~identical, it would be good to move it to the test utility package if possible.

Reply:
done